### DISCUSSION TITLE
porting this project to another controller.

### DISCUSSION THREAD

**Discussion by bacshoreline** [InfiniTimeOrg/InfiniTime#1492] [Q&A]
Hi,
i am thinking to make similar watch with some additional features.This project is written for nrf52832. i am thinking of using nrf52840 with the source code.Can some one guide me for porting this project to nrf52840?What are the required files to be changed?
TIA.

**Comment by JF002** [comment]
Hello,

I have actually already experimented with InfiniTime on the NRF52840 and managed to make it work fairly easily. I can't remember the exact changes I had to make in the code, though.
IIRC I had to edit a few variables in the CMake files   (https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/src/CMakeLists.txt and https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/cmake-nRF5x/CMake_nRF5x.cmake), a few changes in the linker script (https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/gcc_nrf52-mcuboot.ld) and adapting the GPIO initialization to the new HW.
